### PR TITLE
ENH: Expose symlog scaling in plotting API

### LINKED ISSUE #24867: ENH: Expose symlog scaling in plotting API
The default log-scaled axes, activated by the `logx`, `logy`, and `loglog` methods to the Pandas plotting API, do the straightforward thing and take the log of 0 values. It then attempt to plot with these infinite logs, and makes the entire plot unusable without warning in the presence of 0s.

For example:

```python
draws = pd.DataFrame({'freq': np.random.zipf(1.7, 1000) - 1})
draws['rank'] = (-draws['freq']).rank()
draws.plot(x='rank', y='freq', kind='scatter', loglog=True)
```

Matplotlib provides another scale, the `symlog` scale, that makes a small region near 0 linear to avoid these problems. For quick-and-dirty 'look at my data on a log axis' plotting, `symlog` is significantly more useful.

I can access it like this:

```
draws = pd.DataFrame({'freq': np.random.zipf(1.7, 1000) - 1})
draws['rank'] = (-draws['freq']).rank()
p = draws.plot(x='rank', y='freq', kind='scatter', loglog=True)
p.set_xscale('symlog')
p.set_yscale('symlog')
p
```

Either making the `symlog` scale the default log scale for plotting, or supporting a `loglog='sym'` option, would make it significantly easier to do quick data inspection with Pandas' convenience plotting.

### PR DESCRIPTION
**Pull request by charlesdong1991**
- [ ] closes #24867 
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by WillAyd** [comment]
Not terribly familiar with this code but looks OK to me at a glance. 

cc @TomAugspurger - too late to get this in for v0.24 at this point right?

File: pandas/plotting/_core.py [hunk 1]
```diff
@@ -310,9 +310,15 @@ def _setup_subplots(self):
         axes = _flatten(axes)
 
         if self.logx or self.loglog:
-            [a.set_xscale('log') for a in axes]
```
Comment (TomAugspurger) [inline pandas/plotting/_core.py]: Probably slightly cleaner to set

```python
scale = 'symlog if self.sym else 'log'
[a.set_xscale(scale) for a in axes]
```

**Review by TomAugspurger — COMMENTED** [review]
It's not clear to be that this is working correctly. You should add a test that includes an assertion that `ax.get_yscale() == 'symlog'`.

What's the user-facing API? I was thinking something like

```
df.plot(logy="sym")
```

so that a user can get regular log with `log=True`, symlog with `log="symlog"` or "sym", not sure which. And maybe accept "logit" too.

**Comment by charlesdong1991** [comment]
Thanks @TomAugspurger for your review, appreciate a lot!
how about this, default value for `self.log` is `False`, which refers to linear. And user can specify any kind of log they want, e.g. `df.plot(log='logy')` or `df.plot(log='symloglog')`, i think this could be quite straightforward to users.
```python
        if self.log:
            scale_log = ["logx", "logy", "loglog"]
            scale_symlog = ["symlogx", "symlogy", "symloglog"]

            if self.log in scale_log:
                [a.set_xscale(scale_log) for a in axes]
            elif self.log in scale_symlog:
                [a.set_xscale(scale_symlog) for a in axes]
            else:
                raise ValueError("Invalid input for log.")
```
Will be very happy to hear your opinions!

**Comment by mdekstrand** [comment]
I like @TomAugspurger's proposed UI - it's exactly what I had in mind. Since the documented API is already `logx=True`, allowing `logx='sym'` to turn on symlog makes a lot of sense to me as an API user.

**Comment by charlesdong1991** [comment]
Thank you very much @mdekstrand , it's good to know! since my proposal will be quite significant API change, so i would like to hear all of yours opinions first before submitting the pr... i will then keep Tom's proposal first in it!

**Review by jreback — CHANGES_REQUESTED** [review]
pls add a whatsnew note in 0.25.0 other enhancements

**Comment by charlesdong1991** [comment]
sorry, since i have several branches working on the same time... there are some git issues that i didn't notice so i messed it up... i will close this PR, and reopen another one... sorry...

**Pull request closed without merging**